libraries/helpers.rb

```diff
@@ -30,6 +30,37 @@ def keys_from_url(url)
       end
     end
 
+    # Determine of a shell exists on disk
+    #
+    # @return [Boolean]
+    def shell_exists(shell_path)
+      File.exist?(shell_path)
+    end
+
+    # Determines if the user's shell is valid on the machine, otherwise
+    # returns the default of /bin/sh
+    #
+    # @return [String]
+    def default_shell(shell)
```

Comment:
So this method should ideally take no args and just return the default shell for a platform. So based on the logic you are trying to achieve you would return `/bin/sh` for all platforms except AIX. I would also rename this argument-less method to `default_shell_for_platform`.